test/distributed/fsdp/test_checkpoint_wrapper.py

```diff
@@ -184,12 +183,18 @@ def test(use_checkpointing, use_wrapper, use_reentrant):
                 use_checkpointing,
                 use_wrapper=use_wrapper,
                 use_reentrant=use_reentrant,
-            ).cuda()
-            x = torch.randn(10000, 256, requires_grad=True).cuda()
-            torch.cuda.reset_peak_memory_stats()
-            loss = a(x).sum()
-            loss.backward()
-            return torch.cuda.max_memory_allocated()
+            ).to(device)
+            x = torch.randn(10000, 256, requires_grad=True).to(device)
+            if TEST_HPU:
```

Comment:
As @kwen2501  mentioned in one of the comments torch.get_device_module(device.type).reset_peak_memory_stats() should work. We just need to check once if there is any issue with Out-of-tree device like HPU. I just checked it seems to be working : 
```
>>> device=torch.device("hpu:0")
>>> torch.get_device_module(device.type).is_available()
True
```

Reply:
I like this change. It’s a good idea to use `torch.get_device_module(...)` instead of relying on `if...else` statements with device-specific namespaces like `torch.cuda`, `torch.hpu`, or `torch.xpu`.